tools/jdbc/src/jni/duckdb_java.cpp

```diff
@@ -490,7 +490,13 @@ JNIEXPORT jobject JNICALL Java_org_duckdb_DuckDBNative_duckdb_1jdbc_1execute(JNI
 				continue;
 			} else if (env->IsInstanceOf(param, J_String)) {
 				auto param_string = jstring_to_string(env, (jstring)param);
-				duckdb_params.push_back(Value(param_string));
+				try {
+					duckdb_params.push_back(Value(param_string));
+				} catch (Exception const &e) {
+					delete res_ref;
```

Comment:
Unrelated to this PR, but perhaps we can turn `res_ref` into a unique ptr to avoid having to delete it manually in all these different branches?

Reply:
Will that work with it being passed back into the java interface?

Reply:
You can use `.release()` on a unique pointer to turn it back into a raw pointer